components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -257,7 +272,8 @@ interface ChatUIPage {
   OnAPIResponseError(APIError error);
   // Usually the model is changed from the UI client, but occassionally
   // it could be changed from the browser.
-  OnModelChanged(string model_key);
+  // Occassionally, when the browser changes the model, it will also provide a new list of models.
```

Comment:
Not sure I follow this comment - why is it only occassionally and is this when any model changes?

Reply:
a model change commonly happens when a user switches models. in this case, there isn’t an updated list, but we still send it using the OnModelChanged API. 

in instances where the current model needs to be switched programmatically, such as when the current model is removed, we provide a new list. this is what ‘occasionally’ refers to